### ISSUE TITLE
error MSB3086: Task could not find "sgen.exe", seems load a wrong version

### ISSUE THREAD

**Issue by zhangxin511** [microsoft/dotnet-framework-docker#323]
I am using `mcr.microsoft.com/dotnet/framework/sdk:4.8`, got following errors:
```
GenerateSerializationAssemblies target) -> 
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(3533,5): error MSB3086: Task could not find "sgen.exe" using the SdkToolsPath "" or the registry key "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\NETFXSDK\4.7.2\WinSDK-NetFx40Tools-x86". Make sure the SdkToolsPath is set and the tool exists in the correct processor specific location under the SdkToolsPath and that the Microsoft Windows SDK is installed 
```

But run following command it gives 
```
λ docker run --rm mcr.microsoft.com/dotnet/framework/sdk:4.8 where sgen.exe
C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\sgen.exe
```

Seems the Microsoft.Common.CurrentVersion.targets load the SdkToolsPath  wrong?

**Comment by mthalman** [comment]
Same sort of issue as #313

**Comment by MichaelSimons** [comment]
This should be addressed with VS 16.3.  Leaving open to validate.
